eth/chains/base.py

```diff
@@ -227,11 +239,14 @@ def get_canonical_block_by_number(self, block_number: BlockNumber) -> BaseBlock:
         raise NotImplementedError("Chain classes must implement this method")
 
     @abstractmethod
-    def get_canonical_block_hash(self, block_number):
+    def get_canonical_block_hash(self, block_number: BlockNumber) -> Hash32:
         raise NotImplementedError("Chain classes must implement this method")
 
     @abstractmethod
-    def build_block_with_transactions(self, transactions, parent_header):
+    def build_block_with_transactions(self,
+                                      transactions: Tuple[BaseTransaction, ...],
+                                      parent_header: BlockHeader=None
+                                      ) -> Tuple[BaseBlock, Tuple[Receipt, ...], Tuple[BaseComputation, ...]]:      # noqa: E501
```

Comment:
I think our preference in cases like this is to move the entire `self` one line down to have more horizontal space and get rid of the `# noqa: E501`